packages/boot/test/bootstrapTests/test-walletSurvivesZoeRestart.ts

```diff
@@ -65,96 +44,30 @@ const setup: LiquidationSetup = {
   },
 };
 
-const outcome = {
-  bids: [
-    {
-      payouts: {
-        Bid: 0,
-        Collateral: 8.897786,
-      },
-    },
-    {
-      payouts: {
-        Bid: 0,
-        Collateral: 10.01001,
-      },
-    },
-    {
-      payouts: {
-        Bid: 10.46,
-        Collateral: 16.432903,
-      },
-    },
-  ],
-  reserve: {
-    allocations: {
-      ATOM: 0.309852,
-      STARS: 0.309852,
-    },
-    shortfall: 0,
-  },
-  vaultsSpec: [
-    {
-      locked: 3.373,
-    },
-    {
-      locked: 3.024,
-    },
-    {
-      locked: 2.792,
-    },
-  ],
-  // TODO match spec https://github.com/Agoric/agoric-sdk/issues/7837
-  vaultsActual: [
-    {
-      locked: 3.525747,
-    },
-    {
-      locked: 3.181519,
-    },
-    {
-      locked: 2.642185,
-    },
-  ],
-} as const;
-//#endregion
-
 test.before(async t => {
   t.context = await makeLiquidationTestContext(t);
 });
+
 test.after.always(t => {
   return t.context.shutdown && t.context.shutdown();
 });
 
-// Reference: Flow 1 from https://github.com/Agoric/agoric-sdk/issues/7123
-const checkFlow1 = async (
-  t: ExecutionContext<LiquidationTestContext>,
-  {
-    collateralBrandKey,
-    managerIndex,
-  }: { collateralBrandKey: string; managerIndex: number },
-  _expected: any,
-) => {
+test.serial('wallet survives zoe null upgrade', async t => {
   // fail if there are any unhandled rejections
   process.on('unhandledRejection', (error: Error) => {
     t.fail(error.message);
   });
+  const collateralBrandKey = 'ATOM';
+  const managerIndex = 0;
+
+  const { walletFactoryDriver, setupVaults, controller, buildProposal } =
+    t.context;
 
-  const {
-    advanceTimeBy,
-    advanceTimeTo,
-    check,
-    priceFeedDrivers,
-    readLatest,
-    walletFactoryDriver,
-    setupVaults,
-    placeBids,
-    controller,
-    buildProposal,
-  } = t.context;
   const { EV } = t.context.runUtils;
 
-  const buildAndExecuteProposal = async packageSpec => {
+  const buyer = await walletFactoryDriver.provideSmartWallet('agoric1buyer');
+
+  const buildAndExecuteProposal = async (packageSpec: string) => {
```

Comment:
Out of scope but we should move this to SwingsetTestKit

Reply:
take care to pass I/O access explicitly when doing so.